add hybrid to 1.10 upgrade guide (#7195)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/core-upgrade/05-upgrading-to-v1.10.md

Large data sets can slow down dbt build times, making it harder for developers to test new code efficiently. The [`--sample` flag](/docs/build/sample-flag), available for the `run` and `build` commands, helps reduce build times and warehouse costs by running dbt in sample mode. It generates filtered refs and sources using time-based sampling, allowing developers to validate outputs without building entire models.

### Integrating dbt Core artifacts with dbt Cloud projects

With [hybrid projects](/docs/deploy/hybrid-projects), <Constant name="core"/> users working in the command line interface (CLI) can execute runs that seamlessly upload [artifacts](/reference/artifacts/dbt-artifacts) into <Constant name="cloud"/>. This enhances hybrid <Constant name="core"/>/<Constant name="cloud"/> deployments by:

> - Fostering collaboration between dbt <Constant name="cloud"/> + <Constant name="core"/> users by enabling them to visualize and perform [cross-project references](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref) to models defined in <Constant name="core"/> projects. This feature unifies <Constant name="cloud"/> + <Constant name="core"/> workflows for a more connected <Constant name="dbt"/> experience.

Author: ```suggestion
- Fostering collaboration between <Constant name="cloud"/> + <Constant name="core"/> users by enabling them to visualize and perform [cross-project references](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref) to models defined in <Constant name="core"/> projects. This feature unifies <Constant name="cloud"/> + <Constant name="core"/> workflows for a more connected <Constant name="dbt"/> experience.
```

> - Fostering collaboration between dbt <Constant name="cloud"/> and <Constant name="core"/> users by enabling them to visualize and perform [cross-project references](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref) to models defined in <Constant name="core"/> projects. This feature unifies <Constant name="cloud"/> and <Constant name="core"/> workflows for a more connected <Constant name="dbt"/> experience.

Reviewer: Could we change:
"This feature unifies <Constant name="cloud"/> and <Constant name="core"/> workflows for a more connected <Constant name="dbt"/> experience."

To:
"This feature unifies <Constant name="cloud"/> + <Constant name="core"/> workflows for a more connected <Constant name="dbt"/> experience."

Author: @reubenmc do you mean this feature unifies core + cloud workflows..?

- Giving <Constant name="cloud"/> and <Constant name="core"/> users insights into their models and assets in [<Constant name="explorer"/>](/docs/collaborate/explore-projects). To view <Constant name="explorer"/>, you must have have a [developer or read-only license](/docs/cloud/manage-access/seats-and-users).
- (Coming soon) Enabling users working in the [<Constant name="visual_editor"/>](/docs/cloud/visual-editor) to build off of models already created by a central data team in <Constant name="core"/> rather than having to start from scratch.

> Hybrid projects are available as a private beta to [dbt <Constant name="cloud"/> Enterprise accounts](https://www.getdbt.com/pricing). Contact your account representative to register your interest in the beta.

Author: ```suggestion
Hybrid projects are available as a private beta to [<Constant name="cloud"/> Enterprise accounts](https://www.getdbt.com/pricing). Contact your account representative to register your interest in the beta.
```

- (Introduced, disabled by default) [`validate_macro_args`](/reference/global-configs/behavior-changes#macro-argument-validation). If the flag is set to `True`, dbt will raise a warning if the argument `type` names you've added in your macro YAMLs don't match the argument names in your macro or if the argument types aren't valid according to the [supported types](/reference/resource-properties/arguments#supported-types).

> ### Integrating dbt Core artifacts with dbt Cloud projects

Reviewer 2: can we move this up - so that the "Managing changes to legacy behaviors" heading is the last one in this section?

Author: no worries!

Author: moved it up!

> With [Hybrid projects](/docs/deploy/hybrid-projects), dbt Core users can seamlessly upload [artifacts](/reference/artifacts/dbt-artifacts) like [run_results.json](/reference/artifacts/run-results-json), [manifest.json](/reference/artifacts/manifest-json), [catalog.json](/reference/artifacts/catalog-json), [sources.json](/reference/artifacts/sources-json), and so on &mdash; into dbt Cloud after executing a run in the dbt Core command line interface (CLI), which helps:

Reviewer 3: ```suggestion
With [hybrid projects](/docs/deploy/hybrid-projects), dbt Core users working in the command line interface (CLI) can execute runs that seamlessly upload [artifacts](/reference/artifacts/dbt-artifacts) like [run_results.json](/reference/artifacts/run-results-json), [manifest.json](/reference/artifacts/manifest-json), [catalog.json](/reference/artifacts/catalog-json), [sources.json](/reference/artifacts/sources-json), and more into dbt Cloud. This enhances hybrid dbt Core/Cloud deployments by:
```

Reviewer: Is there a reason to specifically list the artifacts here? The ones that get uploaded are the ones that are needed to power features in Cloud, so this may change in the future and is not user controlled.

Author: removed the actual artifacts list

> - Collaborate with dbt Cloud users by enabling them to visualize and perform [cross-project references](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref) to dbt models that live in Core projects.

Reviewer 3: ```suggestion
- Fostering collaboration with dbt Cloud users by enabling them to visualize and perform [cross-project references](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref) to dbt models in Core projects.
```

Reviewer: How about "dbt Cloud users and Core users"?

Also, any chance we can get a One dbt in here? Or is that too much for docs?

Author: - Fostering collaboration between dbt Cloud and Core users by enabling them to visualize and perform [cross-project references](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref) to models defined in Core projects. This feature unifies Cloud and Core workflows for a more connected dbt experience.

Reviewer: Ok if I'm not allowed Onedbt can I have "Core + Cloud" instead of "Cloud and Core"? :)

> - (Coming soon) New users interested in the [Visual Editor](/docs/cloud/visual-editor) can build off of dbt models already created by a central data team in dbt Core rather than having to start from scratch.

Reviewer 3: ```suggestion
- (Coming soon) Enabling users working in the [Visual Editor](/docs/cloud/visual-editor) to build off of source dbt models created by a central data team in dbt Core rather than having to start from scratch.
```

Reviewer: Isn't "source" here confusing given that a dbt DAG has both "sources" and "models" so while they are kind of using the model as a "source" isn't not actually a source in dbt venaculiar. The model is more of the basis for starting out with development via a cross project reg rather than starting with a blank slate.

Author: @reubenmc i agree and wouldn't usually call them source models. happy to keep it as is!

> - dbt Core users can navigate to [dbt Explorer](/docs/collaborate/explore-projects) and view their models and assets. To view dbt Explorer, you must have a read-only seat.

Reviewer 3: ```suggestion
- Giving dbt Core users insights into their models and assets in [dbt Explorer](/docs/collaborate/explore-projects). To view dbt Explorer, users must have a [developer or read-only license](/docs/cloud/manage-access/seats-and-users).
```

Reviewer: This should be for both Core and Cloud users right? Yes Core users can now go in and see them. And so can Cloud users who now want to utilize these Core data products.

> Hybrid projects is available in private beta to [dbt Cloud Enterprise accounts](https://www.getdbt.com/pricing). To register your interest in the beta, reach out to your account representative.

Reviewer 3: ```suggestion
Hybrid projects are available as a private beta to [dbt Cloud Enterprise accounts](https://www.getdbt.com/pricing). Contact your account representative to register your interest in the beta.
```
